vividus-engine/src/main/java/org/vividus/annotation/ReplacementMonitor.java

```diff
@@ -61,22 +61,25 @@ public void stepMatchesPattern(String step, boolean matches, StepPattern pattern
     @Override
     public void beforePerforming(String step, boolean dryRun, Method method)
     {
-        getAnnotation(method).ifPresent(r -> {
-            String rawDeprecatedStep = runTestContext.getRunningStory().getRunningSteps().getFirst();
-            Matcher stepMatcher = getStepMatcher(rawDeprecatedStep);
-            String[] stepParams = getParametersFromStep(stepMatcher);
+        getAnnotation(method).ifPresent(r -> LOGGER.info(
+                getLogMessageForRunningDeprecatedStep(r.versionToRemoveStep(), r.replacementFormatPattern())));
+    }
 
-            try (Formatter newStep = new Formatter())
-            {
-                newStep.format(r.replacementFormatPattern(), stepParams);
+    public String getLogMessageForRunningDeprecatedStep(String versionToRemove, String replacementFormatPattern)
```

Comment:
it should be moved to some other class, and then it should be reused in both StoryReporter and StepMonitor

Reply:
> it should be moved to some other class, and then it should be reused in both StoryReporter and StepMonitor

We need gathered data from this monitor to get deprecation messaged
Probably it would better to left this logic here, but move logging part to separate monitor